http-server-netty/src/main/java/io/micronaut/http/server/netty/multipart/NettyStreamingFileUpload.java

```diff
@@ -212,4 +159,67 @@ public void subscribe(Subscriber<? super PartData> s) {
     public void discard() {
         fileUpload.release();
     }
+
+    private Publisher<Boolean> transferTo(ThrowingSupplier<OutputStream, IOException> outputStreamSupplier) {
+        return Mono.<Boolean>create(emitter ->
+
+                subject.subscribeOn(Schedulers.fromExecutorService(ioExecutor))
```

Comment:
Seems this should be `publishOn` as well like https://github.com/micronaut-projects/micronaut-core/pull/6120/files#diff-5e0dd1237da04622a601621ff253c396e94d189c262f0f9975c89540878ba5bbR98 ?

Reply:
subscribeOn should be sufficient as it applies from the source and down the chain of operators. I verified the thread is indeed IO when the write occurs